src/components/EventBlocks/index.tsx

```diff
@@ -0,0 +1,268 @@
+        }
+        onBlur={
+          canBeTabFocused
+            ? (e: React.FocusEvent): void => {
+                e.stopPropagation();
+                setIsHovered(false);
+              }
+            : undefined
+        }
+        tabIndex={canBeTabFocused ? 0 : -1}
+      >
+        {includeContent && (
+          <div className="meeting-wrapper">
+            <div className="ids">
+              <span className="event-id">{name}</span>
+            </div>
+            {includeTime && (
+              <span className="period">{periodToString(period)}</span>
+            )}
+          </div>
+        )}
+      </BlockElement>
+      {/* Include a details popover that can be opened by either:
+      - hovering over the meeting, on devices with hover. This opens a
+        "preview", which has less opacity.
+      - clicking, tapping, or focusing on the meeting. This "selects" it,
+        shows the popover with more opacity, and adds a selection 'halo' around
+        the meeting block itself. */}
+      {includeDetailsPopover && (
+        <ReactTooltip
+          anchorId={tooltipId}
+          className="tooltip"
+          variant="dark"
+          place="top"
+          clickable
+          isOpen={isSelected || isHovered}
+          setIsOpen={setIsHovered}
+          delayShow={20}
+          delayHide={100}
+          // HACK: ReactTooltip can't handle changing the `events` prop, so we
+          // need to force a remount when the `deviceHasHover` prop changes.
+          // This probably won't happen often in the wild, but it's noticeable
+          // when debugging in devtools:
+          key={deviceHasHover ? 0 : 1}
+          // Disable the hover event if the device doesn't support it, since
+          // clicks are handled separately and present a better UX.
+          // Without this, the hover event will be triggered in the same way as
+          // clicks, which is redundant. Moreover, upon loosing focus on mobile,
+          // the "delayHide" causes the tooltip to remain open for a small
+          // amount of time which is noticeable, especially because the
+          // "selected" styles stop applying while the tooltip is still open.
+          events={deviceHasHover ? ['hover'] : []}
+        >
+          <DetailsPopoverContent name={name} period={period} days={days} />
+        </ReactTooltip>
+      )}
+    </div>
+  );
+}
+
+type DetailsPopoverContentProps = {
+  name: string;
+  period: Period;
+  days: string[];
+};
+
+function DetailsPopoverContent({
+  name,
+  period,
+  days,
+}: DetailsPopoverContentProps): React.ReactElement {
+  return (
+    <table className="details">
+      <tbody>
+        <tr>
+          <td>
+            <b>Name</b>
+          </td>
+          <td>{name}</td>
+        </tr>
+        <tr>
+          <td>
+            <b>Time</b>
+          </td>
+          <td>{[days.join(''), periodToString(period)].join(' ')}</td>
+        </tr>
+      </tbody>
+    </table>
+  );
+}
```

Comment:
One of the only differences between `EventBlocks` and `TimeBlocks`. Refer to the first comment.